src/commands.cpp

```diff
@@ -8136,12 +8137,15 @@ CommandBackupPut::CommandBackupPut(MegaClient *client, BackupType type, handle n
     arg("n", backupName.c_str());
     arg("s", state);
     arg("ss", subState);
-    arg("e", extraData.c_str());
+
+    if (!extraData.empty())
+        arg("e", extraData.c_str());
 
     tag = client->reqtag;
     mUpdate = false;
 }
 
+// to update an already registered backup
 CommandBackupPut::CommandBackupPut(MegaClient* client, handle backupId, BackupType type, handle nodeHandle, const char* localFolder, const char *deviceId, const char* backupName, int state, int subState, const char* extraData)
 {
```

Comment:
Not related to recent commits, but I noticed that updating the state is done in the following way:
```
    if (state > 0)
    {
        arg("s", state);
    }
```
while specs say
```
"s":0,    // sync's state (0: enabled, 1: disabled)
```
which looks like it cannot go _back_ to `enabled` state.

Reply:
very valid indeed. The same should apply to substate. Nice catch @catalinr-m . I'll change to >=0